### ISSUE TITLE
Test Triangulation package with clang

### ISSUE THREAD

**Issue by mglisse** [CGAL/cgal#3605]
Most tests in the Triangulation package start with a check that disables them for old versions of gcc. Sadly, this check also matches clang, no matter how recent, so we are missing some testing there...

`test_delaunay.cpp`:
```c++
#if defined(__GNUC__) && defined(__GNUC_MINOR__) && (__GNUC__ <= 4) && (__GNUC_MINOR__ < 4)
```

**Comment by afabri** [comment]
@maxGimeno can you please fix this.

**Comment by mglisse** [comment]
The test in NewKernel_d seems to use BOOST_GCC to detect the real gcc, it probably just needs to be copied.

**Comment by lrineau** [comment]
Fixed by #3654.
